src/numeric_index.c

```diff
@@ -489,20 +435,21 @@ bool NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
 }
 
 NRN_AddRv NumericRangeTree_TrimEmptyLeaves(NumericRangeTree *t) {
-  NRN_AddRv rv = {.sz = 0, .changed = 0, .numRecords = 0, .numRanges = 0};
+  NRN_AddRv rv = {0};
   NumericRangeNode_RemoveChild(&t->root, &rv);
   if (rv.changed) {
     // Update the NumericTree
     t->revisionId++;
     t->numRanges += rv.numRanges;
-    t->emptyLeaves = 0;
+    t->emptyLeaves += rv.numLeaves;
+    t->numLeaves += rv.numLeaves;
```

Comment:
TODO: NumericRangeTree should hold its stats as struct and move its reference as a bookkeeping context to the functions, to avoid this copy

Reply:
Good idea. I can add it to the story for improving the parent range feature